packages/opentelemetry-context-async-hooks/package.json

```diff
@@ -56,7 +56,7 @@
     "typescript": "4.4.4"
   },
   "peerDependencies": {
-    "@opentelemetry/api": ">=1.0.0 <1.10.0"
+    "@opentelemetry/api": ">=1.3.0 <1.10.0"
```

Comment:
I'm sorry for being so confusing all the time - it's kind of difficult to articulate what the way forward for this is as there's so many moving parts :disappointed: - we're just allowed to bump the API version for all packages **except** for `./packages`, as these are all stable and that's the only place where it's actually breaking and requires us to bump to 2.0. So what you could actually do is bump all versions in 
- `./selenium-tests/` (not public anyway -> API bump is okay)
- `./experimental/` (not stable - therefore we can break in minor -> API bump is okay)
- `./integration-tests` (not public anyway -> API bump is okay)
- `./examples/` (they're all examples, not published to NPM -> API bump is okay)

What we cannot do on `main` is bump the peer-dependency of any package in `./packages` (with one exception, see below) because it's breaking AND the packages are 1.x (semver stable) already. So a user with an older version of `@opentelemetry/api` installed will get an `npm` error and won't be able to typescript compile anymore once we exchange `SpanAttributes` with `Attributes` as the type does not exist on their version of the API.

However, there are a few packages where we can actually exchange the `SpanAttributes` and `MetricAttributes` for `Attributes` on `main` even if the package is `1.x` (semver stable) already, and that's in all packages where we use `@opentelemetry/api@>=1.3.0` already, because:
- we don't need to bump to a later API version (not bumping means we're not breaking)
- `Attributes` already exists in `1.3.0`, and `MetricAttributes` is a type-alias for `Attributes` so it's actually a drop-in replacement that won't cause any compile issues.

Everything else that does not fit these constraints must be done in the `next` branch as they'll actually be breaking for a stable package. :sweat_smile: 

So, to summarize, we can bump and replace `SpanAttributes` and `MetricAttributes` on `main` everywhere **except** for:
- `@opentelemetry/context-async-hooks` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/context-zone` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/context-zone-peer-dep` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/core` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/exporter-jaeger` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/exporter-zipkin` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/propagator-b3` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/propagator-jaeger` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/resources` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/sdk-trace-base` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/sdk-trace-node` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/sdk-trace-web` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/propagator-aws-xray` is 1.x and depends on API >=1.0.0 (no `Attributes` type)
- `@opentelemetry/shim-opentracing` is 1.x and depends on API >=1.0.0 (no `Attributes` type)

These changes would have to go into the `next` branch and will be released with 2.0.

There is a special case with `@opentelemetry/sdk-metrics` which is 1.x but it's already depending on API >=1.3.0, so the `Attributes` type is available. Since that type is available (and there's not difference between `Attributes` and `MetricAttributes` we can just replace all usages of `MetricAttributes` to `Attributes` in `@opentelemetry/sdk-metrics` without any repercussions. :slightly_smiling_face:

Reply:
Hey @pichlermarc 

No worries and thanks for the detailed explanation. I will adjust this PR and use this comment as a guide for the following ones :)

Reply:
I've reverted [1b6f5d53](https://github.com/open-telemetry/opentelemetry-js/pull/4992/commits/1b6f5d536c0615636003a665b4e759dcd564c6f4). So nothing inside `packages` is modified :)

Reply:
Thanks :slightly_smiling_face:

Reply:
@pichlermarc this breakdown was so helpful! I added it to the [original issue as a comment](https://github.com/open-telemetry/opentelemetry-js/issues/4175#issuecomment-2354025623), though it could probably even go in the issue description